SPHINXsys/src/shared/particle_dynamics/solid_dynamics/fluid_structure_interaction.h

```diff
@@ -41,150 +41,129 @@ namespace SPH
 	{
 		typedef DataDelegateSimple<SolidParticles> SolidDataSimple;
 		typedef DataDelegateContact<SolidParticles, FluidParticles> FSIContactData;
+
 		/**
-		 * @class FluidViscousForceOnSolid
-		 * @brief Computing the viscous force from the fluid
+		 * @class BaseForceFromFluid
+		 * @brief Base class for computing the forces from the fluid
 		 */
-		class FluidViscousForceOnSolid : public LocalDynamics, public FSIContactData
+		class BaseForceFromFluid : public LocalDynamics, public FSIContactData
 		{
 		public:
-			explicit FluidViscousForceOnSolid(BaseContactRelation &contact_relation);
-			virtual ~FluidViscousForceOnSolid(){};
-			void interaction(size_t index_i, Real dt = 0.0);
-			StdLargeVec<Vecd> &getViscousForceFromFluid() { return viscous_force_from_fluid_; };
+			explicit BaseForceFromFluid(BaseContactRelation &contact_relation);
+			virtual ~BaseForceFromFluid(){};
+			StdLargeVec<Vecd> &getForceFromFluid() { return force_from_fluid_; };
 
 		protected:
 			StdLargeVec<Real> &Vol_;
-			StdLargeVec<Vecd> &vel_ave_;
 			StdVec<Fluid *> contact_fluids_;
-			StdVec<StdLargeVec<Real> *> contact_rho_n_;
-			StdVec<StdLargeVec<Vecd> *> contact_vel_n_;
-			StdVec<Real> mu_;
-			StdVec<Real> smoothing_length_;
-			StdLargeVec<Vecd> viscous_force_from_fluid_;
+			StdLargeVec<Vecd> force_from_fluid_;
 		};
 
 		/**
-		 * @class FluidViscousForceOnSolidInEuler
-		 * @brief Computing the viscous force from the fluid in eulerian framework
+		 * @class ViscousForceFromFluid
+		 * @brief Computing the viscous force from the fluid
 		 */
-		class FluidViscousForceOnSolidInEuler : public LocalDynamics, public FSIContactData
+		class ViscousForceFromFluid : public BaseForceFromFluid
```

Comment:
seem not suitable for multi-phase flow

Reply:
For this, please open an issue on prof. Hu's repo for him. Why not?